app/services/doc_auth/lexis_nexis/error_generator.rb

```diff
@@ -56,48 +59,89 @@ class ErrorGenerator
+            return { GENERAL => [general_error(false)] }
+          end
+        end
+
+        errors.transform_values(&:to_a)
+      end
+      # rubocop:enable Metrics/PerceivedComplexity
+
+      def self.get_error_messages(liveness_enabled, response_info)
         errors = Hash.new { |hash, key| hash[key] = Set.new }
 
         if response_info[:DocAuthResult] != 'Passed'
-          response_info[:Alerts]&.each do |alert|
+          response_info[:Alerts][:failed]&.each do |alert|
             alert_msg_hash = TRUE_ID_MESSAGES[alert[:name].to_sym]
 
             if alert_msg_hash.present?
-              # Don't show alert errors if the DocAuthResult has passed
-              # With liveness turned on DocAuthResult can pass but the liveness check fails
-              if alert[:result] != 'Passed'
-                # We should log the counts of failing alerts that are given to users for analytics
-                # AM: Log to Ahoy/Cloudwatch
-                errors[alert_msg_hash[:type]].add(I18n.t(alert_msg_hash[:msg_key]))
-              end
-              # We always want to make sure any unknown alerts that come through are noted
-              # AM: Log to Ahoy/Cloudwatch
+              errors[alert_msg_hash[:type]].add(I18n.t(alert_msg_hash[:msg_key]))
             end
           end
-        elsif liveness_checking_enabled && response_info[:PortraitMatchResults].present?
-          # Only bother to look for selfie_results if ID Auth was successful
-          if response_info[:PortraitMatchResults].dig(:FaceMatchResult) != 'Pass'
-            errors[:selfie].add(I18n.t('doc_auth.errors.lexis_nexis.selfie_failure'))
-            # Should probably log if selfie results isn't populated when we expect it to be?
-          end
         end
 
-        if errors.empty?
-          e = UnknownTrueIDError.new('TrueID failure escaped without useful errors')
-          # AM: Log to Ahoy/Cloudwatch
-          NewRelic::Agent.notice_error(e, { custom_params: { response_info: response_info } })
+        pm_results = response_info[:PortraitMatchResults]
+        if liveness_enabled && pm_results.present? && pm_results.dig(:FaceMatchResult) != 'Pass'
+          errors[SELFIE].add(I18n.t('doc_auth.errors.lexis_nexis.selfie_failure'))
+        end
 
-          if liveness_checking_enabled
-            errors[:general].add(I18n.t('doc_auth.errors.lexis_nexis.general_error_liveness'))
-          else
-            errors[:general].add(I18n.t('doc_auth.errors.lexis_nexis.general_error_no_liveness'))
-          end
+        errors
+      end
+
+      def self.general_error(liveness_enabled)
+        if liveness_enabled
+          I18n.t('doc_auth.errors.lexis_nexis.general_error_liveness')
+        else
+          I18n.t('doc_auth.errors.lexis_nexis.general_error_no_liveness')
+        end
+      end
+
+      def self.scan_for_unknown_alerts(response_info)
+        all_alerts = response_info[:Alerts][:failed] + response_info[:Alerts][:passed]
```

Comment:
Is it confirmed both of these will be arrays and not nil? If there's a chance one is nil, we can do:

```suggestion
        all_alerts = [*response_info[:Alerts][:failed], *response_info[:Alerts][:passed]]
```

Reply:
It's being  created in the parsing code and should always be at worst empty arrays and not nil but I like guarding regardless because you never know what the future will bring. Though I'm not 100% certain what this syntax does. I'll try googling.

Reply:
It's splatting:

```
a = [1, 2, 3]
b = 4
c = nil
d = [*a, *b, *c]
# => [1, 2, 3, 4]
```

Reply:
I hadn't known it  could  be used in that way, it's interesting.